pkg/agent/client.go

```diff
@@ -420,12 +420,12 @@ func (a *Client) Serve() {
 						resp.GetCloseResponse().Error = err.Error()
 					}
 
+					close(dataCh)
```

Comment:
I like the change but I think we can do this even more aggressively. If we close the channel and remove the connContext just before line 418, I think it stops us from sending data toward a connection we have already closed.

Reply:
P.S. I think both lines can move up to before ```err := conn.Close()```.